README.md

```diff
@@ -4,35 +4,47 @@
 
 ## Building the CLI
 
-If you simply want to build a binary locally, run
-```sh
-$ make build/turnkey
-```
-
-We use [GoReleaser](https://goreleaser.com/) to build and release our binaries.
-
-To build a release locally:
+If you simply want to build binaries locally, run
 ```
-$ goreleaser release --snapshot --rm-dist
+$ make
 ```
 
 To release:
 ```
-# Generate a Github token with "write:packages"
-# ==> https://github.com/settings/tokens
-$ export GITHUB_TOKEN=<your token>
-$ git tag -s -a vx.y.z -m "New release: x.y.z" # create a signed tag
-$ git tag -v vx.y.z # verify the tag
-$ git push origin vx.y.z
-$ goreleaser release --rm-dist
+make dist
 ```
 
 ## Installing the CLI
```

Comment:
Tried using this and ran into a bunch of issues:

1. Because of git submodules, `make attest` won't work. We need to make git submodule update/init part of the Makefile

2. The current makefile assumes users have `gpg` installed. This needs to be checked and a proper error message needs to be displayed if a user doesn't have `gpg` available or doesn't have the right version

3. When running `make verify` I see the following errors:
```
cat: cache/toolchain.state: No such file or directory
cat: cache/toolchain.state: No such file or directory
cat: cache/toolchain.state: No such file or directory

Verifying: dist/manifest.36C8AAA9.asc

gpg: Signature made Sat Mar 18 02:02:51 2023 CET
gpg:                using RSA key 67553FBDA46BB71ABD2E0B0B8E47A1EC35A1551D
gpg: Can't check signature: No public key
make: *** [verify] Error 2
```

I'm guessing `cache/toolchain.state: No such file or directory` isn't a fatal error, but we should make sure it's not present. We don't want to train users to ignore errors in `make verify`'s output.

The `Can't check signature: No public key` is the "real" error, not quite sure what triggers it.

4. Running `make attest` also yields a lot of `cat: cache/toolchain.state: No such file or directory` in the output.

5. Building (within `make attest`) took 152.8 seconds. This is a long time!

6. `make attest` failed with `groupadd: GID '20' already exists` (when trying to run `docker run --rm --tty --interactive --env UID=501 --env GID=20 --platform=linux/x86_64 --privileged --cpus 5 --volume /Users/rno/tkhq/code/tkcli:/home/build --workdir /home/build --env-file=config/global.env --env-file=cache/toolchain.env sha256:16bab33fc6ac040562cee29bc318881653a861d182253be2ec44a46d58aba140 src/toolchain/scripts/host-env bash -c ' GOHOSTOS="linux" GOHOSTARCH="amd64" GOOS="linux" GOARCH="386" GOCACHE=/home/build/cache/x86_64 GOPATH=/home/build/cache/x86_64 env -C src go build -o /home/build/out/turnkey.linux-386 main.go '`)

Reply:
> 1. Because of git submodules, make attest won't work. We need to make git submodule update/init part of the Makefile

It is in the makefile and automatically runs for me. If it does not for you let me know what version of GNU Make you are running. It would not shock me if MacOS uses a very old version we have to find workarounds for.

https://github.com/tkhq/tkcli/blob/lrvick/deterministic-builds/Makefile#L3-L5

> 2. The current makefile assumes users have gpg installed. This needs to be checked and a proper error message needs to be displayed if a user doesn't have gpg available or doesn't have the right version

I am not aware of any version of MacOS from the past 10 years or so, or any Linux distro, that does not ship with a version that can do "gpg --verify". If I am mistaken here lmk and I can address

> 3. When running make verify I see the following errors:

Fair. I'll add a make hook to fetch and pin all public keys for easy verification.

> 4. Running make attest also yields a lot of cat: cache/toolchain.state: No such file or directory in the output.

Fair. These were introduced in order to make mac builds run faster, but it is noisy. A workaround was not immediately obvious but I'll kick it out to a script if I have to.

> 5. Building (within make attest) took 152.8 seconds. This is a long time!

Make attest must do a from scratch build with no cache. This is just the time it takes to build the container image and run go build in it for each architecture with whatever number of cores you have assigned to Docker.

FWIIW it is only 70 seconds for me.

The same process for QoS takes 10-20m, so I consider this fast!

> 6. make attest failed with groupadd: GID '20' already exists

Ah. How annoying. MacOS apparently abuses user/group ids below 1000 which pretty much every other unix system reserves.  In this case the reserved gid '20' belongs to "dialout" in Linux. I can add logic to toolchain clobber any conflicting user/groups in the container at runtime.

Reply:
@lrvick I double-checked, and as far as I can tell `gpg` doesn't come pre-installed on MacOS. See https://support.apple.com/en-us/HT201214 and https://mikeross.xyz/gpg-without-gpgtools-on-mac/ for examples of install instructions